### PR TITLE
Bump llvm/clang to 15.0.2

### PR DESCRIPTION
**Pull request by azat**
Otherwise right now codebrowser image ~is broken~ cannot be rebuilt [1], due to changes in cmake:

    #10 1.607 CMake Error at /usr/lib/llvm-15/lib/cmake/llvm/LLVMExports.cmake:1693 (message):
    #10 1.607   The imported target "merge-fdata" references the file
    #10 1.607
    #10 1.607      "/usr/lib/llvm-15/bin/merge-fdata"

  [1]: https://s3.amazonaws.com/clickhouse-test-reports/42349/36502270eb9f195dcd2dc8b350db9367c619f0f8/push_to_dockerhub_amd64.html

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by azat** [comment]
After this I was going to update Dockerfile for codebrowser.

File: docker/test/util/Dockerfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ FROM ubuntu:20.04
 ARG apt_archive="http://archive.ubuntu.com"
 RUN sed -i "s|http://archive.ubuntu.com|$apt_archive|g" /etc/apt/sources.list
 
+# 15.0.2
```
Comment (alexey-milovidov) [inline docker/test/util/Dockerfile:8]: Does it mean - you simply force rebuilding the docker image?

**Review by alexey-milovidov — APPROVED** [review]

File: docker/test/util/Dockerfile (hunk 1 above)
Comment (azat) [reply docker/test/util/Dockerfile:8]: Yes

**Comment by azat** [comment]
Blocked by https://github.com/llvm/llvm-project/issues/58317

**Comment by azat** [comment]
>[Stateless tests (debug) [1/3] — Invalid check_status.tsv]https://s3.amazonaws.com/clickhouse-test-reports/42351/123fe3e5dfcf899e52e234ee7ab8bf5d30f28f06/stateless_tests__debug__[1/3].html)

```
docker: error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create": EOF.
```

>[Stress test (asan) — Backward compatibility check: Error message in clickhouse-server.log (see bc_check_error_messages.txt)](https://s3.amazonaws.com/clickhouse-test-reports/42351/123fe3e5dfcf899e52e234ee7ab8bf5d30f28f06/stress_test__asan_.html)

```
2022.10.18 07:43:49.403144 [ 737762 ] {} <Error> virtual void DB::StorageBuffer::flush(): Code: 269. DB::Exception: Destination table is myself. Will lead to infinite loop. (INFINITE_LOOP), Stack trace (when copying this message, always include the lines below):
```

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@azat what do you think about trying Bolt with ClickHouse?

**Comment by azat** [comment]
> what do you think about trying Bolt with ClickHouse?

Usually it is a problem of a good profile report, the thing that good for one may be not good for others.
Maybe simply using profile report after running all tests will work better.
But this will require to run additional build after all tests to optimize the binary.
I guess this can eliminate some random changes in generated code that leads to performance decrease though.

But all this require careful look on the profile and at BOLT itself.

**Pull request merged**